graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/TopSecondDegreeByCountTweetRecsGenerator.java

```diff
@@ -109,6 +107,67 @@ private static boolean isLessThantMinUserSocialProofSize(
     return !keep;
   }
 
+  private static boolean socialProofUnionSizeLessThanMin(
+    SmallArrayBasedLongToDoubleMap[] socialProofs,
+    int minUserSocialProofSize,
+    Set<byte[]> socialProofTypeUnions
+  ) {
+    boolean keep = false;
+    long socialProofSizeSum;
+
+    outerloop:
+    for (byte[] socialProofTypeUnion: socialProofTypeUnions) {
+      socialProofSizeSum = 0;
+      for (byte socialProofType: socialProofTypeUnion) {
+        if (socialProofs[socialProofType] != null) {
+          socialProofSizeSum += socialProofs[socialProofType].size();
+          if (socialProofSizeSum >= minUserSocialProofSize) {
+            keep = true;
+            break outerloop;
+          }
+        }
+      }
+    }
+
+    return !keep;
+  }
+
+  private static boolean isLessThanMinUserSocialProofSizeCombined(
+    SmallArrayBasedLongToDoubleMap[] socialProofs,
+    int minUserSocialProofSize,
+    Set<byte[]> socialProofTypeUnions
+  ) {
+    boolean keep = false;
```

Comment:
could you change the flag name from keep to lessThan in this file and initialize it to true at the start? At the end of the function, we will return lessThan instead of !keep. I started to use keep as the flag name, and I realize now it was a mistake. My mind blows up reading all those negation signs.